### PR TITLE
Use separate property for public key type

### PR DESCRIPTION
**Pull request by alejandro-isaza**
Also expose public key type enum on the interface.

### CODE REVIEW

File: android/app/src/androidTest/java/com/trustwallet/core/app/blockchains/CoinAddressDerivationTests.kt [hunk 1]
```diff
@@ -61,7 +61,7 @@ class CoinAddressDerivationTests {
         DOGECOIN -> assertEquals("DJRFZNg8jkUtjcpo2zJd92FUAzwRjitw6f", address)
         KIN -> assertEquals("GBL3MT2ICHHM5OJ2QJ44CAHGDK6MLPINVXBKOKLHGBWQDVRWTWQ7U2EA", address)
         VIACOIN -> assertEquals("via1qnmsgjd6cvfprnszdgmyg9kewtjfgqflz67wwhc", address)
-        IOST -> assertEquals("TyEiLWJ2R5fN12dCH5i6fmALoYUsn49CqbRhZU96R1GD", address)
+        IOST -> assertEquals("AjvLqiYRd2vbRwqn3uXoUYMpZ3qZ2r57MYUNhYiKu41w", address)
```
Comment (hewigovens) [inline android/app/src/androidTest/java/com/trustwallet/core/app/blockchains/CoinAddressDerivationTests.kt]: Why is IOST testcase changed?

File: tests/Binance/SignerTests.cpp [hunk 2]
```diff
@@ -90,6 +90,26 @@ TEST(BinanceSigner, Build) {
             "1240""2a78b6d9a108eb9440221802b626e24d80179395ac984f016db012ef1a0c16d71b4d7053e05366ae3ea2681fc8052398fda20551c965d74c5970bbc66b94b48e"
             "1801"
             "200a"
+//          fc01
```
Comment (hewigovens) [inline tests/Binance/SignerTests.cpp]: let's remove this

File: android/app/src/androidTest/java/com/trustwallet/core/app/blockchains/CoinAddressDerivationTests.kt (hunk 1 above)
Comment (hewigovens) [reply android/app/src/androidTest/java/com/trustwallet/core/app/blockchains/CoinAddressDerivationTests.kt]: nvm, PublicKey holds 32 bytes now

File: tests/Binance/SignerTests.cpp (hunk 2 above)
Comment (alejandro-isaza) [reply tests/Binance/SignerTests.cpp]: Fixed

File: android/app/src/androidTest/java/com/trustwallet/core/app/blockchains/CoinAddressDerivationTests.kt (hunk 1 above)
Comment (alejandro-isaza) [reply android/app/src/androidTest/java/com/trustwallet/core/app/blockchains/CoinAddressDerivationTests.kt]: Yes, I think it was a bug.

**Pull request merged**